libc-test/build.rs

```diff
@@ -662,6 +665,10 @@ fn main() {
         } else {
             cfg.header("linux/fcntl.h");
         }
+        if !musl {
```

Comment:
This should not have been necessary because these constants aren't tested in this binary. Only your changes to the "main" test binary should have been necessary. Did you try this PR without this change first and only added it once you saw tests failing?